src/common/actions/repository.js

```diff
@@ -78,10 +86,12 @@ export function resetRepository(id) {
   };
 }
 
-function createWebhook({ owner, name }) {
+function createWebhook({ owner, name, id }) {
   return {
+    payload: { id },
     [CALL_API]: {
       path: '/api/github/webhook',
+      types: [REPO_ADD_REQUEST, REPO_ADD_SUCCESS],
```

Comment:
Adding repo actions to webhook actions allows us to show spinner also when we wait for webhook API.